src/dspbase.jl

```diff
@@ -163,37 +197,63 @@ function _conv(u, v, paddims)
     ifft!(upad)
 end
 
-_conv_clip!(y::AbstractVector, minpad) = sizehint!(resize!(y, minpad[1]), minpad[1])
-_conv_clip!(y::AbstractArray, minpad) = y[CartesianIndices(minpad)]
+function _conv_clip!(
+    y::AbstractVector,
+    minpad,
+    ::NTuple{<:Any, Base.OneTo{Int}},
+    ::NTuple{<:Any, Base.OneTo{Int}}
+)
+    sizehint!(resize!(y, minpad[1]), minpad[1])
+end
+function _conv_clip!(
+    y::AbstractArray,
+    minpad,
+    ::NTuple{<:Any, Base.OneTo{Int}},
+    ::NTuple{<:Any, Base.OneTo{Int}}
+)
+    y[CartesianIndices(minpad)]
+end
+# For arrays with weird offsets
+function _conv_clip!(y::AbstractArray, minpad, axesu, axesv)
+    out_offsets = first.(axesu) .+ first.(axesv)
+    out_axes = range.(out_offsets, out_offsets .+ minpad .- 1)
+    out = similar(y, out_axes)
+    copyto!(out, CartesianIndices(out), y, CartesianIndices(range.(1, minpad)))
+end
 
 """
     conv(u,v)
 
 Convolution of two arrays. Uses FFT algorithm.
 """
-function conv(u::StridedArray{T, N}, v::StridedArray{T, N}) where {T<:BLAS.BlasFloat, N}
+function conv(u::AbstractArray{T, N},
+              v::AbstractArray{T, N}) where {T<:BLAS.BlasFloat, N}
     su = size(u)
     sv = size(v)
     minpad = su .+ sv .- 1
     padsize = map(n -> n > 1024 ? nextprod([2,3,5], n) : nextpow(2, n), minpad)
     y = _conv(u, v, padsize)
-    _conv_clip!(y, minpad)
+    _conv_clip!(y, minpad, axes(u), axes(v))
 end
-
-conv(u::StridedArray{T, N}, v::StridedArray{T, N}) where {T<:Integer, N} =
+function conv(u::AbstractArray{<:BLAS.BlasFloat, N},
+              v::AbstractArray{<:BLAS.BlasFloat, N}) where N
+    fu, fv = promote(u, v)
+    conv(fu, fv)
+end
+conv(u::AbstractArray{T, N}, v::AbstractArray{T, N}) where {T<:Number, N} =
+    conv(float(u), float(v))
+conv(u::AbstractArray{T, N}, v::AbstractArray{T, N}) where {T<:Integer, N} =
```

Comment:
This should probably be expanded to have `T1<:Integer` and `T2<:Integer` so it covers when `u` and `v` are both `Integer` types but they're not the same type. (or maybe it should be handled with promotion?) For example this currently throws a `MethodError`:

```
conv([1, 2, 5], Int32[6, 1, 4, 1])
```